src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -1659,11 +2115,793 @@ End Class
+
+            var s1b = @"
+public class Mid : Base
+{
+    public override string P => null;
+}
+";
+            var ref1b = CreateCompilationWithCovariantReturns(s1b, references: new[] { ref0 }, assemblyName: "ref1").EmitToImageReference();
+
+            var s2 = @"
+public class Derived : Mid
+{
+    public override string P => null;
+}
+";
+            var comp = CreateCompilationWithoutCovariantReturns(s2, references: new[] { ref0, ref1a }).VerifyDiagnostics(
+                // (4,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override string P => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "P").WithArguments("covariant returns").WithLocation(4, 28)
+                );
+            verify1(comp);
+
+            comp = CreateCompilationWithCovariantReturns(s2, references: new[] { ref0, ref1a }).VerifyDiagnostics(
+                );
+            verify1(comp);
+            verify1(CompilationReferenceView(comp, ref0, ref1a));
+            verify1(MetadataView(comp, ref0, ref1a));
+            verify1(RetargetedView(comp, ref0, ref1a));
+
+            verify2(CompilationReferenceView(comp, ref0, ref1b));
+            verify2(MetadataView(comp, ref0, ref1b));
+            verify2(RetargetedView(comp, ref0, ref1b));
+
+            static void verify1(CSharpCompilation comp)
+            {
+                VerifyOverride(comp, "Derived.P", "System.String Derived.P { get; }", "System.Object Base.P { get; }");
+                VerifyOverride(comp, "Derived.get_P", "System.String Derived.P.get", "System.Object Base.P.get");
+            }
+
+            static void verify2(CSharpCompilation comp)
+            {
+                VerifyOverride(comp, "Derived.P", "System.String Derived.P { get; }", "System.String Mid.P { get; }", binaryCompat: true);
+                VerifyOverride(comp, "Derived.get_P", "System.String Derived.P.get", "System.String Mid.P.get", binaryCompat: true);
+                VerifyOverride(comp, "Mid.P", "System.String Mid.P { get; }", "System.Object Base.P { get; }");
+                VerifyOverride(comp, "Mid.get_P", "System.String Mid.P.get", "System.Object Base.P.get");
+            }
+        }
+
+        [Fact]
+        public void BinaryCompatibility_03()
+        {
+            var s0 = @"
+public class Base
+{
+    public virtual object M() => null;
+}
+";
+            var ref0 = CreateCompilationWithoutCovariantReturns(s0).EmitToImageReference();
+
+            var s1a = @"
+public class Mid : Base
+{
+}
+";
+            var ref1a = CreateCompilationWithoutCovariantReturns(s1a, references: new[] { ref0 }, assemblyName: "ref1").EmitToImageReference();
+
+            var s1b = @"
+public class Mid : Base
+{
+    public override object M() => null;
```

Comment:
>public override object M() => null; [](start = 4, length = 35)

I think we should also test scenario when this is:
- new virtual method, I assume it will be newSlot
- new but not virtual, I assume it will be newSlot
- all of the above including original but when there is no hidebysig flag in metadata.

Reply:
These suggestions are recorded in the test plan.

---
In reply to: [424783253](https://github.com/dotnet/roslyn/pull/44025#discussion_r424783253) [](ancestors = 424783253)